### DISCUSSION TITLE
rspamd and ipv6 redis

### DISCUSSION THREAD

**Discussion by adjustGirl** [rspamd/rspamd#3701] [General]
Rspamd doesn't work several times a day and I have to restart again.  Why does rspamd access the ipv6?  in the redis.conf there is only 127.0.0.1.

2021-04-03 23:55:36 #4078(controller) rspamdredispoolnewconnection: cannot connect to redis ::1 (port 6379): Die angeforderte Adresse kann nicht zugewiesen werden
2021-04-03 23:55:36 #4078(controller) rspamdluaredisprepareconnection: cannot connect to redis (OS error): Die angeforderte Adresse kann nicht zugewiesen werden
2021-04-03 23:55:36 #4078(controller) lua; lua_redis.lua:1072: cannot execute redis request

Where do I have to look?  Or configure ...

In worker-controller.inc?

**Comment by moisseev** [comment]
`rspamadm configdump redis.servers`
`rspamadm configdump | grep -C 3 servers`

**Reply by adjustGirl** [reply]
redis {
    write_servers = "localhost";
    timeout = 120;
    read_servers = "localhost";
    servers = "127.0.0.1";
}
milter_headers {
    authenticated_headers [
--
        extended = true;
        type = "spamassassin";
        symbol = "SPAMD";
        servers = "127.0.0.1:783";
    }
}
dkim_signing {
--
                }
            }
            read_only = true;
            servers = "round-robin:fuzzy1.rspamd.com:11335,fuzzy2.rspamd.com:11335";
            algorithm = "mumhash";
            min_length = 64;
            max_score = 20;

**Reply by adjustGirl** [reply]
[dump.txt](https://github.com/rspamd/rspamd/files/6254459/dump.txt)

**Comment by moisseev** [comment]
```ucl
redis {
    servers = "127.0.0.1";
    write_servers = "localhost";
    timeout = 120;
    read_servers = "localhost";
}
```
Commonly `/etc/hosts` maps `localhost` to both IPv4 and IPv6 addresses.
You should remove the `write_servers = "localhost";` and ` read_servers = "localhost";` as you don't use distinct read and write Redis servers.

**Reply by adjustGirl** [reply]
Thank you, work perfectly
